wicket-util/src/main/java/org/apache/wicket/util/string/Strings.java

```diff
@@ -1089,7 +1089,7 @@ public static CharSequence toMultilineMarkup(final CharSequence s)
 			return null;
 		}
 
-		final AppendingStringBuffer buffer = new AppendingStringBuffer();
+		final AppendingStringBuffer buffer = new AppendingStringBuffer((int) (s.length() * 1.1));

```

Comment:
This looks like black magic :)
Maybe comment and/or meaningful constant name should be added? :)

Reply:
We have the exact same logic in `Strings.escapeMarkup`:

https://github.com/apache/wicket/blob/d51883862a448e26f8d973a24756ea7e03259de7/wicket-util/src/main/java/org/apache/wicket/util/string/Strings.java#L305

A constant would have to be something like `ESTIMATED_OVERHEAD_OF_MULTILINE_MARKUP`. I think a comment would be better.

Reply:
Would it make sense to have this "magic" in some setting?

Reply:
I don't think so. The additional 10% should be enough to accommodate markup and escaping overhead in most cases. And if it isn't enough, the buffer is resized once. This is *much* better than resizing the buffer almost every time as it is now. Imho there is no need to configure or optimize this further.

Reply:
> I don't think so. The additional 10% should be enough to accommodate markup and escaping overhead in most cases. And if it isn't enough, the buffer is resized once. This is _much_ better than resizing the buffer almost every time as it is now. Imho there is no need to configure or optimize this further.

Yes... I after I wrote my comment I looked at soucre code and I realized the 10% more should be enough. Maybe what we can have is a TenPercentMoreAppendingStringBuffer class?

Reply:
I just realized that this has an adverse effect on short strings where the overhead of `<p></p>` is significantly more than 10%. It makes sense to set a minimum capacity for the buffer. Either via `Math.max((int)(len * 1.1), 16)` or by always adding the default capacity of 16 to the length.